wagtail_wordpress_import/importers/wordpress.py

```diff
@@ -173,9 +181,27 @@ def run(self, *args, **kwargs):
                 )
             self.logger.log_progress()
 
-        self.connect_richtext_page_links(self.imported_pages)
-        if getattr(settings, "WORDPRESS_IMPORT_HOOKS_ITEMS_TO_CACHE", {}):
-            self.items_cache.process(self.imported_pages)
+        imported_pages = self.page_model_instance.objects.filter(
+            id__in=[id for id in self.imported_page_ids]
+        ).specific()
+
+        self.connect_richtext_page_links(imported_pages)
+
+        """Run all hooks in settings.WORDPRESS_IMPORT_HOOKS_ITEMS_TO_CACHE"""
+        for hook, actions in getattr(
+            settings, "WORDPRESS_IMPORT_HOOKS_ITEMS_TO_CACHE", {}
+        ).items():
+            import_string(actions["FUNCTION"])(
+                imported_pages, actions["DATA_TAG"], self.items_cache.__dict__[hook]
+            )
+
+        """Run all hooks in settings.WORDPRESS_IMPORT_HOOKS_TAGS_TO_CACHE"""
+        for hook, actions in getattr(
+            settings, "WORDPRESS_IMPORT_HOOKS_TAGS_TO_CACHE", {}
+        ).items():
+            import_string(actions["FUNCTION"])(
+                imported_pages, actions["DATA_TAG"], self.tags_cache.__dict__[hook]
+            )
```

Comment:
They are cached at different points early on in the for loop. 

TAGS are cached before we have entered an item that is a post or page type `wp:post_type` and ITEMS are cached when we have entered an `<item>` tag. 

I did it this way because we expand the tag (node) to a dict and we are only moving forward each iteration. We should only expand the tag if it's required. At first I thought the expand could happen first but it doesn't work.